database/seeders/MainSeeder.php

```diff
@@ -27,6 +31,25 @@ public function run()
             $mediumData = get_object_vars($x->medium);
             $mediumModel = Medium::firstOrCreate($mediumData);
 
+            //genre
+            $genreData = $x->genre;
+            foreach ($genreData as $genre){
+                $genreModel = Genre::firstOrCreate(['name' => $genre->name]);
```

Comment:
You either add a new genre or fetch a genre named $genre->name its ok, but You don't make a relationship between genre and medium.
Put simply, you fill the genres table but don't add anything to the table genre_medium :( 
It is very important!

here's a tip:
[https://laravel.com/docs/9.x/eloquent-relationships#inserting-and-updating-related-models](url)